File: Android/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/loginfo/LogExportDialog.java
package com.didichuxing.doraemonkit.kit.loginfo;


import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.didichuxing.doraemonkit.R;
import com.didichuxing.doraemonkit.ui.dialog.DialogListener;
import com.didichuxing.doraemonkit.ui.dialog.DialogProvider;
import com.didichuxing.doraemonkit.ui.setting.SettingItem;
import com.didichuxing.doraemonkit.ui.setting.SettingItemAdapter;

import java.io.File;

/**
 * Created by wanglikun on 2019/4/16
 */
public class LogExportDialog extends DialogProvider<Object> {
    private RecyclerView mChooseList;
    private SettingItemAdapter mAdapter;

    public LogExportDialog(Object data, DialogListener listener) {
        super(data, listener);
    }

    @Override
    public int getLayoutId() {
        return R.layout.dk_dialog_file_explorer_choose;
    }

    @Override
    protected void findViews(View view) {
        mChooseList = view.findViewById(R.id.choose_list);
        mAdapter = new SettingItemAdapter(getContext());
        mChooseList.setAdapter(mAdapter);
        mChooseList.setLayoutManager(new LinearLayoutManager(getContext()));
    }

    @Override
    protected void bindData(final Object file) {
        mAdapter.append(new SettingItem(R.string.dk_save));

        mAdapter.append(new SettingItem(R.string.dk_share));
        mAdapter.setOnSettingItemClickListener(new SettingItemAdapter.OnSettingItemClickListener() {
            @Override
            public void onSettingItemClick(View view, SettingItem data) {
                if (data.desc == R.string.dk_save) {
                    if (onButtonClickListener != null) {
                        onButtonClickListener.onSaveClick(LogExportDialog.this);
                    }
                } else if (data.desc == R.string.dk_share) {
                    if (onButtonClickListener != null) {
                        onButtonClickListener.onShareClick(LogExportDialog.this);
                    }
                }
            }
        });
    }

    private OnButtonClickListener onButtonClickListener;

    public void setOnButtonClickListener(OnButtonClickListener onButtonClickListener) {
        this.onButtonClickListener = onButtonClickListener;
    }

    public interface OnButtonClickListener {
        /**
         * @param dialog dialog
         */
        void onSaveClick(LogExportDialog dialog);

        /**
         * @param dialog dialog
         */
        void onShareClick(LogExportDialog dialog);
    }
}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/loginfo/LogExportDialog.kt
package com.didichuxing.doraemonkit.kit.loginfo

import android.view.View
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.didichuxing.doraemonkit.R
import com.didichuxing.doraemonkit.kit.core.SettingItem
import com.didichuxing.doraemonkit.kit.core.SettingItemAdapter
import com.didichuxing.doraemonkit.widget.dialog.DialogListener
import com.didichuxing.doraemonkit.widget.dialog.DialogProvider

/**
 * @author lostjobs created on 2020/6/28
 */
class LogExportDialog(listener: DialogListener? = null) : DialogProvider<List<SettingItem>>(listOf(SettingItem(desc = R.string.dk_save), SettingItem(desc = R.string.dk_share)), listener) {

    var onButtonClickListener: OnButtonClickListener? = null

    private lateinit var mChooseList: RecyclerView
    private val mAdapter: SettingItemAdapter by lazy {
        SettingItemAdapter(context)
    }

    override val layoutId: Int = R.layout.dk_dialog_file_explorer_choose

    override fun findViews(view: View?) {
        val rootView = view ?: return
        mChooseList = rootView.findViewById(R.id.choose_list)
        mChooseList.layoutManager = LinearLayoutManager(context)
        mChooseList.adapter = mAdapter
    }

    override fun bindData(data: List<SettingItem>) {
        mAdapter.addData(data)
        mAdapter.setOnSettingItemClickListener(object : SettingItemAdapter.OnSettingItemClickListener {
            override fun onSettingItemClick(view: View, data: SettingItem) {
                when (data.desc) {
                    R.string.dk_save -> {
                        onButtonClickListener?.onSaveClick(this@LogExportDialog)
                    }
                    R.string.dk_share -> {
                        onButtonClickListener?.onShareClick(this@LogExportDialog)
                    }
                }
            }
        })
    }

    override val isCancellable: Boolean = false

    interface OnButtonClickListener {
        fun onSaveClick(dialog: LogExportDialog)
        fun onShareClick(dialog: LogExportDialog)
    }
}